internal/ansible/proxy/proxy_test.go

```diff
@@ -23,99 +23,55 @@ import (
-	}
-
-	po, err := createPod("test", "default", cl)
-	if err != nil {
-		t.Fatalf("Failed to create the pod: %v", err)
-	}
+var _ = Describe("proxyTests", func() {
+	t := GinkgoT()
 
-	resp, err := http.Get("http://localhost:8888/api/v1/namespaces/default/pods/test")
-	if err != nil {
-		t.Fatalf("Error getting pod from proxy: %v", err)
-	}
-	defer func() {
-		if err := resp.Body.Close(); err != nil && !errors.Is(err, os.ErrClosed) {
-			t.Errorf("Failed to close response body: (%v)", err)
+	It("should retrieve resources from the cache", func() {
+		if testing.Short() {
+			Skip("skipping ansible proxy testing in short mode")
 		}
-	}()
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		t.Fatalf("Error reading response body: %v", err)
-	}
-	// Should only be one string from 'X-Cache' header (explicitly set to HIT in proxy)
-	if resp.Header["X-Cache"] == nil {
-		t.Fatalf("Object was not retrieved from cache")
-		if resp.Header["X-Cache"][0] != "HIT" {
-			t.Fatalf("Cache response header found but got [%v], expected [HIT]", resp.Header["X-Cache"][0])
+		po, err := createTestPod("test", "test-watched-namespace", testClient)
+		if err != nil {
+			t.Fatalf("Failed to create the pod: %v", err)
 		}
-	}
-	data := kcorev1.Pod{}
-	err = json.Unmarshal(body, &data)
-	if err != nil {
-		t.Fatalf("Error parsing response: %v", err)
-	}
-	if data.Name != "test" {
-		t.Fatalf("Got unexpected pod name: %#v", data.Name)
-	}
-	if err := cl.Delete(context.Background(), po); err != nil {
-		t.Fatalf("Failed to delete the pod: %v", err)
-	}
-}
+		defer func() {
+			if err := testClient.Delete(context.Background(), po); err != nil {
+				t.Fatalf("Failed to delete the pod: %v", err)
+			}
+		}()
 
-func createPod(name, namespace string, cl client.Client) (client.Object, error) {
-	three := int64(3)
-	pod := &kcorev1.Pod{
-		ObjectMeta: kmetav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
-			Labels: map[string]string{
-				"test-label": name,
-			},
-		},
-		Spec: kcorev1.PodSpec{
-			Containers:            []kcorev1.Container{{Name: "nginx", Image: "nginx"}},
-			RestartPolicy:         "Always",
-			ActiveDeadlineSeconds: &three,
-		},
-	}
-	if err := cl.Create(context.Background(), pod); err != nil {
-		return nil, err
-	}
-	return pod, nil
-}
+		resp, err := http.Get("http://localhost:8888/api/v1/namespaces/test-watched-namespace/pods/test")
+		if err != nil {
+			t.Fatalf("Error getting pod from proxy: %v", err)
+		}
+		defer func() {
+			if err := resp.Body.Close(); err != nil && !errors.Is(err, os.ErrClosed) {
+				t.Errorf("Failed to close response body: (%v)", err)
+			}
+		}()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("Error reading response body: %v", err)
+		}
+		// Should only be one string from 'X-Cache' header (explicitly set to HIT in proxy)
+		if resp.Header["X-Cache"] == nil {
+			t.Fatalf("Object was not retrieved from cache")
+			if resp.Header["X-Cache"][0] != "HIT" {
```

Comment:
Should this be hoisted to the outer scope, since the header doesn't exist?

Reply:
yeah, I didn't actually modify much of this code, was just trying to finagle it to work better with a suite of tests, looks like it's been wrong for a while, I'll go fix it https://github.com/operator-framework/operator-sdk/pull/5064/files#diff-73191ccf5a8f38e276927084c15465129d38040f4d0d6cdce06d6d0b132d01e3L81-L85